### PR TITLE
Refactor storage interface and implementation

### PR DESCRIPTION
**Pull request by iuwqyir**
### TL;DR

Refactored storage implementation and integrated it with the orchestrator and worker components.

### What changed?

- Consolidated storage interfaces into a single `IStorage` interface
- Removed separate orchestrator storage package and integrated it into the main storage package
- Updated `ClickHouseConnector` and `MemoryConnector` to implement both `IOrchestratorStorage` and `IDBStorage` interfaces
- Modified `Orchestrator`, `Poller`, `FailureRecoverer`, and `Worker` to use the new unified storage interface
- Simplified storage initialization in the `Orchestrator`
- Updated `Worker` to use the new storage interface for data processing
- Removed unused `queryRows` function from `Worker`
- Adjusted import statements and function signatures throughout the codebase to reflect the new storage structure

### CODE REVIEW

**Comment by iuwqyir** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/13?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#13** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/13?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#12** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/12?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iuwqyir and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by AmineAfia — APPROVED** [review]

**Comment by iuwqyir** [comment]
### Merge activity

* **Sep 18, 6:22 AM EDT**: @iuwqyir started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thirdweb-dev/indexer/13).
